stratum/tools/stratum-replay/stratum_replay.cc

```diff
@@ -0,0 +1,229 @@
+    key_materials_config->set_pem_root_certs(pem_root_certs);
+
+    if (!FLAGS_client_cert.empty() && !FLAGS_client_key.empty()) {
+      RETURN_IF_ERROR(::stratum::ReadFileToString(
+          FLAGS_client_cert, &pem_key_cert_pair.cert_chain));
+      RETURN_IF_ERROR(::stratum::ReadFileToString(
+          FLAGS_client_key, &pem_key_cert_pair.private_key));
+      key_materials_config->add_pem_key_cert_pair(pem_key_cert_pair);
+    }
+    auto cred_opts = ::grpc_impl::experimental::TlsCredentialsOptions(
+        GRPC_SSL_DONT_REQUEST_CLIENT_CERTIFICATE, GRPC_TLS_SERVER_VERIFICATION,
+        key_materials_config, nullptr, nullptr);
+    channel_credentials = grpc::experimental::TlsCredentials(cred_opts);
+  }
+  auto channel = ::grpc::CreateChannel(FLAGS_grpc_addr, channel_credentials);
+  auto stub = ::p4::v1::P4Runtime::NewStub(channel);
+
+  // Sends the arbitration update with given device id and election id.
+  ::p4::v1::StreamMessageRequest stream_req;
+  std::vector<std::string> election_ids =
+      absl::StrSplit(FLAGS_election_id, ",");
+  CHECK_RETURN_IF_FALSE(election_ids.size() == 2) << "Invalid election ID.";
+  uint64 election_id_high;
+  uint64 election_id_low;
+  CHECK_RETURN_IF_FALSE(absl::SimpleAtoi(election_ids[0], &election_id_high))
+      << "Unable to parse string " << election_ids[0] << " to uint64";
+  CHECK_RETURN_IF_FALSE(absl::SimpleAtoi(election_ids[1], &election_id_low))
+      << "Unable to parse string " << election_ids[1] << " to uint64";
+  absl::uint128 election_id =
+      absl::MakeUint128(election_id_high, election_id_low);
+
+  stream_req.mutable_arbitration()->set_device_id(FLAGS_device_id);
+  stream_req.mutable_arbitration()->mutable_election_id()->set_high(
+      absl::Uint128High64(election_id));
+  stream_req.mutable_arbitration()->mutable_election_id()->set_low(
+      absl::Uint128Low64(election_id));
+
+  ::grpc::ClientContext context;
+  std::unique_ptr<ClientStreamChannelReaderWriter> stream =
+      stub->StreamChannel(&context);
+  if (!stream->Write(stream_req)) {
+    RETURN_ERROR(ERR_INTERNAL)
+        << "Failed to send request '" << stream_req.ShortDebugString()
+        << "' to switch.";
+  }
+
+  // Push the given pipeline config
+  ::p4::v1::SetForwardingPipelineConfigRequest fwd_pipe_cfg_req;
+  ::p4::v1::SetForwardingPipelineConfigResponse fwd_pipe_cfg_resp;
+  fwd_pipe_cfg_req.set_device_id(FLAGS_device_id);
+  fwd_pipe_cfg_req.mutable_election_id()->set_high(
+      absl::Uint128High64(election_id));
+  fwd_pipe_cfg_req.mutable_election_id()->set_low(
+      absl::Uint128Low64(election_id));
+  fwd_pipe_cfg_req.set_action(
+      ::p4::v1::SetForwardingPipelineConfigRequest::VERIFY_AND_COMMIT);
+
+  ::stratum::hal::ForwardingPipelineConfigs pipeline_cfg;
+  RETURN_IF_ERROR(ReadProtoFromTextFile(FLAGS_pipeline_cfg, &pipeline_cfg));
+  const ::p4::v1::ForwardingPipelineConfig* fwd_pipe_cfg =
+      ::stratum::gtl::FindOrNull(pipeline_cfg.node_id_to_config(),
```

Comment:
```suggestion
      gtl::FindOrNull(pipeline_cfg.node_id_to_config(),
```

Should work, since we're already in the `stratum` namespace.